sxw.c

```diff
@@ -217,21 +218,21 @@ void SXW_Init( Bool init_SW, char *f_roots ) {
  * \ingroup SXW
  */
 static void SXW_Reinit(char* SOILWAT_file) {
-	char *temp;
+	PathInfo.InFiles[eFirst] = strdup(SOILWAT_file);
```

Comment:
Don't we need to free previously allocated memory in `PathInfo.InFiles[eFirst]`? Since `PathInfo` is a global and we may be calling `SXW_Reinit()` multiple time. Are we freeing that memory elsewhere?

Reply:
I'm going to rebuttal this by going off of what I believe STEPWAT2 is doing. Doesn't STEPWAT2 need to deallocate existing SOILWAT2 memory in order to reinitialize, resulting in the resetting of PathInfo.InFiles? Or in other words, shouldn't STEPWAT2 already be correctly resetting it? Meaning either `SXW_Reinit()`/`SXW_Init()` is called once per run, or SOILWAT2 (including the first input file) is reset through `SXW_Reset()`.

Reply:
I agree that STEPWAT2 should already correctly resetting SOILWAT2 memory; currently, however, it does not do that correctly in gridded mode. 

See  `_init_stepwat_inputs()` > loop over gridcells: `_init_SXW_inputs()` > `SXW_Init()` > `SXW_Reinit()` -- since gridcells continue to share global variables `SoilWatAll`, etc. and `SXW_Init()` does not call `SW_CTL_clear_model()`

It looks like we can fix this by having `SXW_Init()` call `SXW_Reset()` instead of `SXW_Reinit()`, see 9689e14100b503e576b5083dcbaba729e1f4fee2